extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/RuntimeRegistry.java

```diff
@@ -30,4 +34,26 @@ public Object unwrap(Object value) {
             ? ((RuntimeValue)value).getValue()
             : value;
     }
+
+    // TODO: fix upstream
```

Comment:
Whats the issue upstream?

Reply:
Sorry was late and forgot to open an issue. The problem is that the default implementations does not merge the results from the repositories with the fallback one so if you bind something to the registry and one of the repositories return a non null set, then you' won't get the local instance but an empty answer.